### ISSUE TITLE
Optional config settings are no longer optional?!?

### ISSUE THREAD

**Issue by jflorian** [yo61/puppet-logrotate#29]
My catalogs recently started failing though I hadn't changed my manifests or Hiera data at all.  The only thing I can think that may be at play here is that my Puppet Master has been updated.  Here's what I'm presently running:

$ rpm -qa | grep puppet
puppet-agent-1.5.0-1.el7.x86_64
puppetserver-2.4.0-1.el7.noarch
puppetdb-terminus-3-1.el7.noarch
puppetlabs-release-7-12.noarch
puppetdb-4.1.0-1.el7.noarch
puppetlabs-release-pc1-1.0.0-2.el7.noarch
puppetdb-termini-4.1.0-1.el7.noarch

The error I'm seeing looks like:
```
Error: Could not retrieve catalog from remote server: Error 400 on SERVER: Evaluation Error: Error while evaluating a Resource Statement, Evaluation Error: Error while evaluating a Function Call, Logrotate::Conf[/etc/logrotate.conf]: compress must be a boolean at /etc/puppetlabs/code/environments/production/forge_modules/logrotate/manifests/conf.pp:61:7 on node foo
```

My manifest has only the following relevant to this module:
```
    include '::logrotate'
```
Everything else happens in Hiera.  I originally had in my common.yaml:
```
logrotate::config:
    compress:               false
    dateext:                true
```

For a few select nodes, I set compress to true instead.  That's it.  It was really trivial and clean.

But now it seems that Puppet may require me to declare every parameter.  I've been adding a parameter for each error I see and so far I'm up to:
```
logrotate::config:
    dateext:                true
    compress:               false
    copy:                   false
    copytruncate:           false
    delaycompress:          false
    missingok:              true
    sharedscripts:          false
    shred:                  false
    ifempty:                false
```
Each new attempt at applying the catalog requires me to specify yet another.

**Comment by robinbowes** [comment]
I suspect this is the same as #26, which is caused by a regression/bug in puppet.

Can you check and, if so, close this issue and track #26?

**Comment by jflorian** [comment]
@robinbowes Thanks for the fast response.  Indeed it is the same.  You just saved me a major hassle of hunting down the relevant Puppet bug.  **Thank you!**

**Comment by robinbowes** [comment]
I just merged a PR which should work around the bug if you need the module fast.

**Comment by jflorian** [comment]
Sorry, my comment above is inaccurate.  I had commented out the `include '::logrotate'` temporarily, then ran into another error elsewhere that had me scratching my head as to how things were working at all.  That's when I got your comment about #26, so I downgraded to puppetserver-2.3.2-1.el7.noarch.  That did nothing for my new bug, but I found that was just a stupid syntax error (with an unquoted string).  With that resolved I jumped to the conclusion that the downgrade resolved this issue, but really the catalog was only apply successfully because the logrotate module was still commented out.  Doh!!!

So now that this module is included again, I'm getting the same errors and this is with puppetserver-2.3.2-1.el7.noarch on the server and puppet-4.2.1-2.fc23.noarch on the node.  I would have thought the downgrade on the server would've resolved this.  I'll try that PR out to see if it helps here.

**Comment by jflorian** [comment]
Okay, hopefully this is something more useful for feedback.  With logrotate-1.3.0 (via `puppet module`) things seem broken with either puppetserver-2.3.2-1.el7.noarch or puppetserver-2.4.0-1.el7.noarch when tested on a node with puppet-4.2.1-2.fc23.noarch.  However, if I replace logrotate-1.3.0 with a git clone whose HEAD is 6b6cfdb (Merge pull request #28 from durist/undef2UNDEFINED) everything is fine with either puppetserver version mentioned here (and with the same node).

**Comment by robinbowes** [comment]
Ok, so that confirms that #28 fixes the issue for you. I'll try and get a new release sorted this weekend so you can still install from the Forge.

R.

**Comment by robinbowes** [comment]
Fixed in 1.4.0

**Comment by jflorian** [comment]
v1.4.0 is working great here.  Thanks again for the zippy response and help!
